### PR TITLE
test: Add playground experiment integration test with all LLM providers

### PR DESCRIPTION
**Pull request by RogerHYang**
## Summary

Adds a comprehensive integration test for the `chatCompletionOverDataset` GraphQL subscription that validates all four LLM provider evaluators work correctly end-to-end.

- Tests OpenAI, Anthropic, Google GenAI, and AWS Bedrock evaluators
- Verifies experiment runs, spans, annotations, and token counts
- Includes bug fixes discovered during testing (AWS Bedrock tool choice, Google streaming null check)

## Test plan

- [x] `tox r -e integration_tests -- -k playground` passes
- [x] `tox r -e ruff,type_check_integration_tests` passes

---

> [!NOTE]
> **Medium Risk**
> Medium risk because it changes AWS Bedrock tool-choice serialization and tweaks Google streaming token accounting, which can affect runtime request payloads and telemetry; the rest is test-only infrastructure and dependency updates.
> 
> **Overview**
> Adds a new integration test suite for the `chatCompletionOverDataset` GraphQL subscription that runs experiments over a dataset and validates runs, traces/spans, token counts, and evaluator annotations across **OpenAI, Anthropic, Google GenAI, and AWS Bedrock** using a new `_mock_llm_server`.
> 
> Fixes provider edge cases uncovered by the test: Bedrock tool-choice is now converted via `AwsToolChoiceConversion.to_aws(...)` (instead of OpenAI conversion), and Google GenAI streaming only updates token-count attributes when `usage_metadata` is present.
> 
> Updates integration/dev dependencies and lockfile to support the new test infrastructure (adds `hypothesis-jsonschema`, Bedrock runtime typing deps, and related AWS async client deps) and ignores `.hypothesis/` artifacts.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 8aa836a760178f2a887e6b51fd72c0b3d83af384. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -93,6 +94,7 @@ dev = [
   "grpcio",
   "httpx",
   "httpx-ws",
+  "hypothesis-jsonschema>=0.23.1",
```
Comment (mikeldking) [inline pyproject.toml:97]: I don't see this used?

Comment (RogerHYang) [reply pyproject.toml:97]: this is used by the mock llm server to generate tool calls from json schemas
